OpenRA.Mods.Common/Traits/SpawnActorsOnSell.cs

```diff
@@ -72,6 +76,33 @@ void Emit(Actor self)
 			}
 
 			var eligibleLocations = buildingInfo.Tiles(self.Location).ToList();
+
+			var guaranteedActorTypes = Info.GuaranteedActorTypes?.Select(a =>
+			{
+				var av = self.World.Map.Rules.Actors[a].TraitInfoOrDefault<ValuedInfo>();
+				return new
+				{
+					Name = a,
+					Cost = av?.Cost ?? 0
+				};
+			}).ToList();
+
+			while (eligibleLocations.Count > 0 && guaranteedActorTypes != null && guaranteedActorTypes.Any())
```

Comment:
Trivial, using both `Count` and `Any` to check Any.
Trivial, place `null` check in parent if block.
This while loop is almost the same as the while loop after. Better to create a function and call it twice.

Reply:
using Count

it's similar but has quite a few differences. Ones that I'm not sure are worth the hassle to account for in a function